ModulePresentationsForCAP/gap/ModulePresentationsForCAP.gi

```diff
@@ -2146,3 +2172,296 @@ end );
     end, 1000 );
 
 end );
+
+##
+## Random methods
+##
+
+##
+## Interpretation of n:
+## The number of rows and colms of the matrix of the object is less or equal to n.
+InstallGlobalFunction( ADD_RANDOM_OBJECT_LEFT,
+
+  function( category )
+
+    AddRandomObject( category,
+
+      function( C, n )
+        local nr_rows, nr_cols, mat, homalg_ring;
+
+        homalg_ring := category!.ring_for_representation_category;
+
+        if n = 0 then
+          
+          return ZeroObject( category );
+
+        fi;
+
+        nr_rows := Random( [ 1 .. n ] );
+
+        nr_cols := Random( [ 1 .. n ] );
+
+        mat := homalg_ring!.random_matrix_func( nr_rows, nr_cols );
+
+        return AsLeftPresentation( mat );
+
+      end );
+
+end );
+
+##
+InstallGlobalFunction( ADD_RANDOM_OBJECT_RIGHT,
+
+  function( category )
+
+    AddRandomObject( category,
+        function( category, n )
+        local homalg_ring, left_presentations, M;
+
+        homalg_ring := category!.ring_for_representation_category;
+
+        left_presentations := LeftPresentations( homalg_ring );
+
+        M := RandomObject( left_presentations, n );
+
+        return AsRightPresentation( Involution( UnderlyingMatrix( M ) ) );
```

Comment:
Suggestion: instead of deriving the random right object from the random left one (which is a bit strange since all you need is an arbitrary random matrix), write a single `ADD_RANDOM_OBJECT` function. This function should have a boolean `left_or_right` as an extra argument (indicating if it is meant to be used for left or right modules), and can call the function you wrote for RandomObject in the left modules case, except that you call the constructor `AsLeftOrRightPresentation( mat, left_or_right)` in the end.

Reply:
Since the derivation of the random morphisms in the right presentations uses the random morphisms in the left presentations, I found it Ok to also derive random objects using the left presentations showing how these two categories relate to each other. Any way, I can change it as you propose. Milliseconds quicker is better than nothing.

Reply:
These changes have been done and included in the PR.

Reply:
Thanks a lot, this change increased the comprehensibility of the code.